src/lib.rs

```diff
@@ -196,21 +208,82 @@ fn do_camera_movement(
     let delta_device_pixels = current_pos - last_pos.unwrap_or(current_pos);
 
     for (cam, mut transform, projection) in &mut query {
-        if !cam.enabled
-            || !cam
-                .grab_buttons
-                .iter()
-                .any(|btn| mouse_buttons.pressed(*btn) && !mouse_buttons.just_pressed(*btn))
-        {
+        if !cam.enabled {
             continue;
         }
 
         let proj_area_size = projection.area.size();
-        let world_units_per_device_pixel = proj_area_size / window_size;
+
+        let mouse_delta = if !cam
+            .grab_buttons
+            .iter()
+            .any(|btn| mouse_buttons.pressed(*btn) && !mouse_buttons.just_pressed(*btn))
+        {
+            None
+        } else {
+            Some(delta_device_pixels * proj_area_size / window_size)
+        };
+
+        let mut direction = Vec2::ZERO;
+
+        if cam
+            .move_keys
+            .left
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
+        {
+            direction.x += 1.;
+        }
+
+        if cam
+            .move_keys
+            .right
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
+        {
+            direction.x -= 1.;
+        }
+
+        if cam
+            .move_keys
+            .up
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
+        {
+            direction.y -= 1.;
+        }
+
+        if cam
+            .move_keys
+            .down
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
+        {
+            direction.y += 1.;
+        }
+
+        let keyboard_delta = if direction.length_squared() == 0. {
```

Comment:
i didn't think of this yesterday, but this is checking that the vec is zeros, right? so `if direction == Vec2::ZERO` would be more readable? unless my brain isn't working :D

i like having one mode like this. it's nice!

Reply:
Yes, I also think a simple zero check would be more readable.

I also wonder if it's better to just let the delta variables be `Vec2` instead of `Option<Vec2>`?

and then simply do:

```rust
let delta = mouse_delta + keyboard_delta;

if delta == Vec2::ZERO {
    continue;
}
```

Would that work? Maybe my brain is also not fully operational :P